machine/dni/dni_3048up/kernel/platform-dni_3048up.patch

```diff
@@ -0,0 +1,1933 @@
+ #elif defined(CONFIG_MACH_HR2)
+ #define BCMIPROC_CHIP_ID	BCM56150_CHIP_ID
+diff --git a/drivers/bcmdrivers/gmac/src/include/typedefs.h b/drivers/bcmdrivers/gmac/src/include/typedefs.h
+index 4c787aa..1ec3ccd 100644
+--- a/drivers/bcmdrivers/gmac/src/include/typedefs.h
++++ b/drivers/bcmdrivers/gmac/src/include/typedefs.h
+@@ -219,7 +219,7 @@ typedef unsigned __int64 uint64;
+ 
+ #endif
+ 
+-#if defined(CONFIG_MACH_NSP) || defined(CONFIG_MACH_HX4)    /* JIRA:LINUXDEV- */
++#if defined(CONFIG_MACH_NSP) || defined(CONFIG_MACH_HX4) || defined(CONFIG_MACH_DNI_3048UP)    /* JIRA:LINUXDEV- */
+ #ifdef CONFIG_CPU_BIG_ENDIAN
+ #define IL_BIGENDIAN
+ #endif /* CONFIG_CPU_BIG_ENDIAN */
+diff --git a/drivers/bcmdrivers/gmac/src/shared/aiutils.c b/drivers/bcmdrivers/gmac/src/shared/aiutils.c
+index d53ade3..a9729a9 100755
+--- a/drivers/bcmdrivers/gmac/src/shared/aiutils.c
++++ b/drivers/bcmdrivers/gmac/src/shared/aiutils.c
+@@ -29,7 +29,7 @@
+ #include <pcicfg.h>
+ 
+ #include "siutils_priv.h"
+-#if defined(CONFIG_MACH_HX4)
++#if defined(CONFIG_MACH_HX4) || defined(CONFIG_MACH_DNI_3048UP)
+ #include "hx4_erom.h"
+ #elif defined(CONFIG_MACH_HR2)
+ #include "hr2_erom.h"
+@@ -61,7 +61,7 @@ get_erom_ent(si_t *sih, uint32 **eromptr, uint32 mask, uint32 match)
+ 	while (TRUE) {
+ #if defined(CONFIG_MACH_NS)
+ 		ent = R_REG(si_osh(sih), *eromptr);
+-#elif defined(CONFIG_MACH_HX4)
++#elif defined(CONFIG_MACH_HX4) || defined(CONFIG_MACH_DNI_3048UP)
+ 		ent = **eromptr;
+ #elif defined(CONFIG_MACH_HR2)
+ 		ent = **eromptr;
+@@ -255,7 +255,7 @@ BCMATTACHFN(ai_scan)(si_t *sih, void *regs, uint devid)
+ 	case SI_BUS:
+ #if defined(CONFIG_MACH_NS)
+ 		eromptr = (uint32 *)REG_MAP(erombase, SI_CORE_SIZE);
+-#elif defined(CONFIG_MACH_HX4)
++#elif defined(CONFIG_MACH_HX4) || defined(CONFIG_MACH_DNI_3048UP)
+ 		eromptr = hx4_erom;
+ #elif defined(CONFIG_MACH_HR2)
+ 		eromptr = hr2_erom;
+diff --git a/drivers/bcmdrivers/gmac/src/shared/nvramstubs.c b/drivers/bcmdrivers/gmac/src/shared/nvramstubs.c
+index aa8c3bd..7707cc2 100755
+--- a/drivers/bcmdrivers/gmac/src/shared/nvramstubs.c
++++ b/drivers/bcmdrivers/gmac/src/shared/nvramstubs.c
+@@ -18,6 +18,8 @@
+  * $Id: nvramstubs.c 325991 2012-04-05 10:16:42Z $
+  */
+ 
++#define FAKE_NVRAM	1
++
+ #include <bcm_cfg.h>
+ #include <typedefs.h>
+ #include <bcmutils.h>
+@@ -62,7 +64,8 @@ static nvram_t fake_nvram[] = {
+ 	{"sdram_config",	"0x103"},
+ 	{"swgmacet",		"et2"},
+ 	{"brcmtag",			"1"},
+-	//{"ethaddr",			"00:90:4c:06:a5:72"},
++	{"ethtool",			"1"},
++	{"ethaddr",			"00:90:4c:06:a5:72"},
+ #ifdef FOUR_PORT_CONFIG
+ 	{"vlan1hwname",		"et2"},
+ 	{"vlan1ports",		"0 1 2 8*"},
+@@ -100,6 +103,8 @@ static nvram_t fake_nvram[] = {
+ #define CONFIG_NAND_BASE		0x1c000000
+ #if (defined(CONFIG_MACH_NS) || defined(CONFIG_MACH_NSP))
+ #define CONFIG_ENV_OFFSET		0xa0000		/* 30000-b0000 - use last 10000 for env */
++#elif defined(CONFIG_MACH_DNI_3048UP)
++#define CONFIG_ENV_OFFSET		0xf0000		/* f0000-fffff - use 10000 for env */
```

Comment:
According to dni_3048up.dts, env_offset is 0x00100000 which doesn't match with this definition.